cmd/create/cluster/cmd.go

```diff
@@ -2332,7 +2332,39 @@ func run(cmd *cobra.Command, _ []string) {
 		isSharedVPC = true
 	}
 
-	if len(subnetIDs) == 0 && isSharedVPC {
+	// validate flags for hcp shared vpc
+	isHcpSharedVpc := false
+	route53RoleArn := sharedVPCRoleARN // TODO: Change when fully deprecating old flag name
+	vpcEndpointRoleArn := strings.Trim(args.vpcEndpointRoleArn, " \t")
+	hcpInternalCommunicationHostedZoneId := strings.Trim(args.hcpInternalCommunicationHostedZoneId, " \t")
+	ingressPrivateHostedZoneId := privateHostedZoneID // TODO: Change when fully deprecating old flag name
+
+	anyHcpSharedVpcFlagsUsed := route53RoleArn != "" || vpcEndpointRoleArn != "" ||
+		hcpInternalCommunicationHostedZoneId != "" || ingressPrivateHostedZoneId != ""
+
+	if anyHcpSharedVpcFlagsUsed {
```

Comment:
imho, it's better to include all these logic into the func validateHcpSharedVpcArgs()